data/scripts.js

```diff
@@ -900,6 +900,127 @@ exports.BattleScripts = {
 
 		return team;
 	},
+	queryMoves: function (moves, hasType, hasAbility) {
+		// This is primarily a helper function for random setbuilder functions.
+		var counter = {
+			Physical: 0, Special: 0, Status: 0, damage: 0, recovery: 0, stab: 0,
+			blaze: 0, overgrow: 0, swarm: 0, torrent: 0,
+			adaptability: 0, ate: 0, bite: 0, contrary: 0, hustle: 0,
+			ironfist: 0, serenegrace: 0, sheerforce: 0, skilllink: 0, technician: 0,
+			inaccurate: 0, priority: 0, recoil: 0,
+			physicalsetup: 0, specialsetup: 0, mixedsetup: 0, speedsetup: 0,
+			damagingMoves: [],
+			damagingMoveIndex: {},
+			setupType: ''
+		};
+
+		if (!moves || !moves.length) return counter;
+		if (hasType === undefined) hasType = false;
+		if (hasAbility === undefined) hasAbility = false;
```

Comment:
If we wanted every last bit of performance we could just get rid of these checks. They should never be true anyway.

While it's true that initializing a proper object is more expensive than a boolean value, property access -which is done a lot of times here- doesn't sound very engine-friendly for a primite boolean. And indeed, access to properties of <code>false</code> is 7x slower than to properties of <code>{}</code>.